src/gurt/errno.c

```diff
@@ -133,3 +142,22 @@ d_errno_deregister_range(int start)
 	D_ERROR("Attempted to deregister non-existent error range from %d\n",
 		start);
 }
+
+const char *
+d_errdesc(int errnum)
+{
+	struct d_error_reg *entry;
+
+	if (errnum == 0)
+		return "Success";
+
+	errnum = D_ABS(errnum);
```

Comment:
My intention was to support the same input range as d_errstr() does. So, should I remove `rc = D_ABS(rc);` from line 93 and 154?

Reply:
I didn't realise that was there but yes.  Given it's already there I'd change that function to issue an error if positive values are called so we can flush out any existing errors.

Reply:
Did you mean to leave this one in?  Looks like you've removed it from d_errstr() but not here, although updated the tests as though you had?

Reply:
It is a bit tricky. All the error codes are stored/registered as positive values (See daos_errno.h L44 – L239). However, d_errdesc(3) and d_errstr(3) expect a negative value. That’s why I have this boilerplate on **both** functions:

`if (errnum > 0) {`
  `  goto out; /* return Unknown error nnn or DER_UNKNOWN */`
`}`
`errnum = D_ABS(errnum);`

From here, we check if the err code is register.

If you prefer, I can change all the error codes on daos_errno.h from positive to negative

Reply:
I was thinking change the loop on L171-L175 to work with negative errnos but positive values in the struct, rather than change the error codes in the header, however what you've done here also works.

Reply:
In that case, may I ask for your approval? :)

Reply:
I thought I had, sorry.